### DISCUSSION TITLE
Subdomain like vercel help needed

### DISCUSSION THREAD

**Discussion by Chetan11-dev** [vercel/community#41] [Help] [answered]
I want to dynamically create subdomain sites such as "user1.example.com", "user2.example.com" and serve static html and css based on subdomain. For a real world example blogger.com servers html and css for different users on blogspot domain link https://althouse.blogspot.com/. I want to do exactly the same thing like blogger.com does.

Questions:
* What code do I need to write for determining subdomain (using js). For example user1.example.com the subdomain is user1.
* Should I store user html, css in s3 by creating folders such as user1/, user2/.

**Comment by steven-tey** [comment]
Hey! Great question – we recently rolled out a [simple example](https://github.com/vercel/examples/tree/main/edge-functions/hostname-rewrites) that would allow you to create unique content pages with a multi-tenant infrastructure. Each of your users gets assigned a unique subdomain when they create their account, with the (usually paid) option to add a custom domain. 

We're also rolling out more complex examples & boilerplates (e-commerce storefront builders, website builders, etc.) in an [early access program](https://eap.vercel.sh/) – feel free to join if you're interested (click "sign up" on the Platforms on Vercel program).

**Reply by omarkhairy21** [reply]
Hi @steven-tey  do i need to assign the custom domain to my app domains in vercel project, let me emphasize that let say my app runs on "_main.com_" in production i will let the user choose custom domain "_custom.com_" for example, do I need to assign this custom domain " _custom.com_ " to my vercel project domains so my app now assigned to these domains [_main.com_ and _example.com_] is it correct ?!

**Reply by steven-tey** [reply]
@omarkhairy21 Yes that's correct! We recently created a boilerplate example on how to do this programmatically using our Domains API: https://domains-api.vercel.app/

And here's the GH repo for that: https://github.com/vercel-customer-feedback/platforms/tree/main/code-snippets/domains-api

Note: You'll need to be part of the [Early Access Program](https://eap.vercel.sh/) to be able to access that repo (click "sign up" on the Platforms on Vercel program).

**Reply by omarkhairy21** [reply]
@steven-tey thanks for replying i am already registered to the Early Access Program but when i go to  [https://github.com/vercel-customer-feedback/platforms/tree/main/code-snippets/domains-api ](url) I got a 404 error also i signup my company name _windr_ on EAP Program list for _Platforms on Vercel_ how could i access the repo ?!

**Reply by steven-tey** [reply]
@omarkhairy21 I just added you to the repo so you should be able to access it now!

**Reply by gr-qft** [reply]
Can you add me as well? Thanks!

**Reply by steven-tey** [reply]
@gr-qft done! :)

**Reply by gr-qft** [reply]
@steven-tey  Thanks!

**Comment by steven-tey** [comment, marked as answer]
Quick update on this – we recently launched the Platforms Starter Kit that's a comprehensive template for building multi-tenant applications with built-in custom domains support (+ unlimited custom domains 😉): https://demo.vercel.pub/platforms-starter-kit

Feel free to try it out and let us know what you think! :)

**Reply by steven-tey** [reply]
CC: @omarkhairy21 @gr-qft @Chetan11-dev
